nautobot/ipam/models.py

```diff
@@ -531,8 +531,19 @@ def __init__(self, *args, **kwargs):
         prefix = kwargs.pop("prefix", None)
         self._location = kwargs.pop("location", None)
         super().__init__(*args, **kwargs)
+
         self._deconstruct_prefix(prefix)
 
+        self._parent = None
+        self._network = None
+        self._prefix_length = None
+        self._namespace = None
+
+        if self.pk:
```

Comment:
using `self.pk` at the `__init__` level instead of `self.present_in_database` because currently thats the only way that works after testing several times.
`self.present_in_database` always returns `False` at `__init__`

Reply:
That's weird... it *should* work when initing from an existing DB record. I'm concerned that checking `self.pk` may lead to false positives, especially after #6340.

Reply:
present_in_database evaluates to false even after calling super().__init__()?

Reply:
Yes

Reply:
Ok, looks like after some reading we might need to instead overload `from_db()` (https://docs.djangoproject.com/en/4.2/ref/models/instances/#customizing-model-loading) and set the cached attributes there instead of in `__init__()`?